demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -119,3 +102,185 @@ InferenceEngine::BlobMap Cnn::Infer(const cv::Mat &frame) {
+        return in_dec_hidden_name_;
+    if (std::find(input_names_decoder.begin(),
+            input_names_decoder.end(),
+            features_name_) == input_names_decoder.end())
+        return features_name_;
+    if (std::find(input_names_decoder.begin(),
+            input_names_decoder.end(),
+            in_dec_symbol_name_) == input_names_decoder.end())
+        return in_dec_symbol_name_;
+    if (std::find(output_names_decoder.begin(),
+            output_names_decoder.end(),
+            out_dec_hidden_name_) == output_names_decoder.end())
+        return out_dec_hidden_name_;
+    if (std::find(output_names_decoder.begin(),
+            output_names_decoder.end(),
+            out_dec_symbol_name_) == output_names_decoder.end())
+        return out_dec_symbol_name_;
+    return std::string("");
+ }
+
+void EncoderDecoderCNN::Init(const std::string &model_path, Core & ie, const std::string & deviceName, const cv::Size &new_input_resolution) {
+    // ---------------------------------------------------------------------------------------------------
+    // --------------------------- Checking paths --------------------------------------------------------
+    std::string model_path_decoder = model_path;
+    auto network_encoder = ie.ReadNetwork(model_path);
+    CNNNetwork network_decoder;
+    try {
+        if (model_path_decoder.find("encoder") == std::string::npos)
+            throw std::runtime_error("Model path does not contain 'encoder'");
+        while (model_path_decoder.find("encoder") != std::string::npos)
+            model_path_decoder = model_path_decoder.replace(model_path_decoder.find("encoder"), 7, "decoder");
+        network_decoder = ie.ReadNetwork(model_path_decoder);
+    }
+    catch (const std::runtime_error& error)
+    {
+        throw std::runtime_error(std::string("Decoder model could not be loaded\n") + error.what());
+    }
+    // --------------------------- Checking net names ----------------------------------------------------
+    InputsDataMap inputInfo(network_encoder.getInputsInfo());
+    if (inputInfo.size() != 1) {
+        throw std::runtime_error("The network_encoder should have only one input");
+    }
+    OutputsDataMap outputInfo(network_encoder.getOutputsInfo());
+    for (auto output : outputInfo) {
+        output_names_encoder.emplace_back(output.first);
+    }
+    outputInfo = network_decoder.getOutputsInfo();
+    for (auto output : outputInfo) {
+        output_names_decoder.emplace_back(output.first);
+    }
+    std::vector<std::string> input_names_decoder;
+    inputInfo = network_decoder.getInputsInfo();
+    for (auto input : inputInfo) {
+        input_names_decoder.emplace_back(input.first);
+    }
+    auto name_not_exist = this->check_net_names(output_names_encoder,
+                                                input_names_decoder,
+                                                output_names_decoder);
```

Comment:
No need to save `output_names_encoder` and `output_names_decoder` as members. This allows you to teach `check_net_names()` to read from `DataMap`s directly.

Reply:
Done